### DISCUSSION TITLE
[General] Question related to Optimism's Regenesis

### DISCUSSION THREAD

**Discussion by gtupak** [ethereum-optimism/developers#608] [General]
### Discussion Topic

Hi everyone,

I am interested to understand 1) how was the USDT contract deployed, because the optimistic etherscan leads to an invalid deploy hash and deployer and 2) what are the GENESIS_-prefixed transactions.

Finally, I've read about the Regenesis [here](https://docs.optimism.io/builders/tools/monitor/regenesis-history) but still don't really understand what happened. Could someone point me in the right direction, please?

Many thanks!

**Comment by Chomtana** [comment]
The GENESIS transaction usually means that the contract bytecode is hardcoded in the `genesis.json` file. There is no transaction or deployer involved in deploying these contracts.

**Reply by gtupak** [reply]
Would you mind giving me a link of this file, or where I can learn about how this works? I looked it up and it wasn't obvious to me. Thank you in advance!

**Reply by Chomtana** [reply]
The Optimism mainnet appears to depend on a hardcoded snapshot and can no longer sync from scratch. As a result, there is no genesis file containing all the contracts.

https://github.com/ethereum-optimism/superchain-registry/pull/2

**Reply by gtupak** [reply]
Hmm.. I see. Do you know where I can learn more about the history of the regenesis, why it happened, etc? This would probably help me answer the two original questions.

**Reply by Chomtana** [reply]
The Regenesis occurred during the Bedrock merge, a major upgrade of the OP mainnet (and OP Goerli). Since Bedrock is essentially a new chain built from scratch and fully incompatible with pre-Bedrock versions, we need to find a way to migrate the state.

Bedrock upgrade proposal is posted at https://gov.optimism.io/t/final-upgrade-1-bedrock-protocol-upgrade-v2/5548

But since it's long time ago, many links doesn't exists anymore
